### PR TITLE
[Blaze i4] Setting local notification for no campaign during 30 days

### PR DESCRIPTION
**Pull request by irfano**
Part of https://github.com/woocommerce/woocommerce-android/issues/12290

### Description

This introduces a new local notification which is a reminder for Blaze. This PR only covers scheduling and displaying the notification. The logic determining when to show the notification will be addressed in the following PR. Tapping the notification should open the campaign creation screen.

### Steps to reproduce

This PR does not trigger the notification. To trigger the notification 31 seconds after launching the app, apply this patch (ensure that notification permissions are enabled): [Trigger_blaze_reminder_notification_for_testing.patch](https://github.com/user-attachments/files/16624484/Trigger_blaze_reminder_notification_for_testing.patch)

### Images/gif

- [x] I have considered adding unit tests for this change. If I decided not to add them, I have provided a brief explanation below (optional):
- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/notifications/local/LocalNotification.kt [hunk 1]
```diff
@@ -15,9 +16,19 @@ sealed class LocalNotification(
     open val data: String? = null
     val id = type.hashCode()
 
-    abstract fun getDescriptionString(resourceProvider: ResourceProvider): String
+    open fun getTitleString(resourceProvider: ResourceProvider) = resourceProvider.getString(title)
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/notifications/local/LocalNotification.kt]: Minor np, but do we need these to be `open` ? I know this was set that way before, but it seems that with the current usage we don't need them to be `open` yet and can probably default the 2 functions to be final.
Wdyt?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/notifications/local/PreconditionCheckWorker.kt [hunk 2]
```diff
@@ -32,13 +31,13 @@ class PreconditionCheckWorker @AssistedInject constructor(
...
         return when (type) {
-            LocalNotificationType.UNUSED -> proceedIfFreeTrialAndMatchesSite(siteId)
+            LocalNotificationType.BLAZE_NO_CAMPAIGN_REMINDER -> proceedIfValidSite(siteId)
 
             null -> cancelWork("Notification type is null. Cancelling work.")
         }
     }
 
-    private fun proceedIfFreeTrialAndMatchesSite(siteId: Long): Result {
+    private fun proceedIfValidSite(siteId: Long): Result {
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/notifications/local/PreconditionCheckWorker.kt:40]: Thanks for catching this piece of legacy naming 👍🏼

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivityViewModel.kt [hunk 3]
```diff
@@ -260,6 +262,11 @@ class MainActivityViewModel @Inject constructor(
                 AnalyticsEvent.LOCAL_NOTIFICATION_TAPPED,
                 mapOf(AnalyticsTracker.KEY_TYPE to notification.tag)
             )
+            LocalNotificationType.fromString(notification.tag)?.let {
+                when (it) {
+                    BLAZE_NO_CAMPAIGN_REMINDER -> triggerEvent(ShowAllCampaigns)
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivityViewModel.kt]: Hmm, unsure if you plan on changing this in subsequent PRs. But haven't seen any updates in the [next PR](https://github.com/woocommerce/woocommerce-android/pull/12374/files). Why open the campaign list screen? Afaik we should open the campaign creation flow: pe5sF9-331-p2

**Review by JorgeMucientes — APPROVED** [review]
Hey @irfano great work! 

I left a couple comments. One of them worth checking before merging eventhough it can be addressed in the [next PR](https://github.com/woocommerce/woocommerce-android/pull/12374). So I'll pre approve the PR but I think we need to update the destination when tapping on the PR. 

Also, I'd like to align with iOS on when they plan on releasing this local reminders. Becasuse we might want to align and release at the same time. In which case we'll need to add feature flag to the notification scheduling.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/notifications/local/LocalNotification.kt (hunk 1 above)
Comment (irfano) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/notifications/local/LocalNotification.kt]: Good catch! I removed `open`s: 4eb9358961ad30e48b2b7684bf4c457ff357ef4e

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivityViewModel.kt (hunk 3 above)
Comment (irfano) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivityViewModel.kt]: Good catch again! Somehow, I thought the campaign list was the campaign creation screen. I've updated it: c30f1d7082e84af944bcaf7325c34555822f2467.

**Comment by irfano** [comment]
Thanks for catching those, @JorgeMucientes. I requested another review because the changes I added were not minor.

> Also, I'd like to align with iOS on when they plan on releasing this local reminders. Becasuse we might want to align and release at the same time. In which case we'll need to add feature flag to the notification scheduling.

Thanks for the reminder. I think we can at least merge this one since this PR doesn’t schedule the notification.

**Review by JorgeMucientes — APPROVED** [review]
Thanks for tackling the changes @irfano. Great job!! Code looks good and works as expected :shipit:

**Pull request merged**